src/PackageVersions/Installer.php

```diff
@@ -80,27 +80,33 @@ public static function getSubscribedEvents()
      */
     public static function dumpVersionsClass(Event $composerEvent)
     {
-        $io = $composerEvent->getIO();
+        $composer = $composerEvent->getComposer();
+        $versions = iterator_to_array(self::getVersions($composer->getLocker(), $composer->getPackage()));
 
-        $io->write('<info>ocramius/package-versions:</info>  Generating version class...');
+        if (!isset($versions['ocramius/package-versions'])) {
```

Comment:
Should probably be an `array_key_exists()` at this point - don't trust the value below it